packages/server/src/course/course.service.ts

```diff
@@ -234,4 +236,63 @@ export class CourseService {
       );
     }
   }
+
+  async getUserInfo(
+    courseId: number,
+    page: number,
+    pageSize: number,
+    role?: Role,
+    search?: string,
+  ): Promise<UserPartial[]> {
+    try {
+      const query = await getRepository(UserModel)
+        .createQueryBuilder()
+        .leftJoin(
+          UserCourseModel,
+          'UserCourseModel',
+          '"UserModel".id = "UserCourseModel"."userId"',
+        )
+        .where('"UserCourseModel"."courseId" = :courseId', { courseId });
+
+      // check if searching for specific role
+      if (role) {
+        query.andWhere('"UserCourseModel".role = :role', { role });
+      }
+      // check if searching for specific name
+      if (search) {
+        const likeSearch = `%${search}%`.toUpperCase();
+        query.andWhere(
+          new Brackets((q) => {
+            q.where('UPPER("UserModel"."firstName") like :firstSearch', {
+              firstSearch: likeSearch,
+            }).orWhere('UPPER("UserModel"."lastName") like :lastSearch', {
+              lastSearch: likeSearch,
+            });
+          }),
+        );
+      }
+
+      // run query
+      const users = query
+        .select([
+          'UserModel.id',
+          'UserModel.firstName',
+          'UserModel.lastName',
+          'UserModel.photoURL',
+          'UserModel.email',
+        ])
+        .orderBy('UserModel.firstName')
+        .skip((page - 1) * pageSize)
```

Comment:
hmm, this isn't a suggestion for change. just food for thought: i suspect having `skip` won't be a problem in sql queries since I would hope there's not that many students/TAs in a course
https://stackoverflow.com/questions/34110504/optimize-query-with-offset-on-large-table/34291099#34291099

Reply:
hmm i see so to summarize, `skip` is fine to use in our case because we don't have too much data in our tables, but if we did it would be very slow. therefore, if we wanted to optimize it we would paginate using `where` queries with a range of `id`'s instead?

Reply:
yessir